### PR TITLE
perf(iroh-net): simplify stun::is check

### PR DESCRIPTION
**Pull request by dignifiedquire**
This is run on every packet, and showed up in some profiling as not being as cheap as it should be

### CODE REVIEW

File: iroh-net/src/stun.rs [hunk 1]
```diff
@@ -65,13 +65,15 @@ pub fn response(tx: TransactionId, addr: SocketAddr) -> Vec<u8> {
     buffer
 }
 
+// Copied from stun_rs
+// const MAGIC_COOKIE: Cookie = Cookie(0x2112_A442);
+const COOKIE: [u8; 4] = 0x2112_A442u32.to_be_bytes();
```
Comment (divagant-martian) [inline iroh-net/src/stun.rs:70]: why the need to copy the constant?

File: iroh-net/src/stun.rs [hunk 2]
```diff
@@ -508,6 +510,11 @@ mod tests {
         assert_eq!(got_tx, tx);
     }
 
+    #[test]
+    fn test_stun_cookie() {
+        assert_eq!(stun_rs::MAGIC_COOKIE, COOKIE);
+    }
```
Comment (divagant-martian) [inline iroh-net/src/stun.rs:516]: we can do this in compile time

File: iroh-net/src/stun.rs (hunk 1 above)
Comment (dignifiedquire) [reply iroh-net/src/stun.rs:70]: because I can't get it out of the library as constant time

File: iroh-net/src/stun.rs (hunk 2 above)
Comment (dignifiedquire) [reply iroh-net/src/stun.rs:516]: the equality check/conversion is not constant time, so not sure how to do that

**Pull request merged**